### PR TITLE
DH-11

### PR DESCRIPTION
**Pull request by Dyrk84**
New Action menu: running away.
Upgraded nextTurn method.
2 Tests to running away action.

### CODE REVIEW

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 1]
```diff
@@ -31,14 +32,26 @@ public Fight(int numOfMonsters) {  //for test
         monsterIncomingOrWin();
     }
 
-    private void monsterIncomingOrWin(){
+    public void runningAway() {
+        System.out.println("The monster hits you a last one before you can run away: ");
```
Comment (gergely-szanto-epam) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]:
```suggestion
        System.out.println("The monster hits you a last time before you can run away: ");
```

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 2]
```diff
@@ -36,5 +36,34 @@ public void championDefeatTest() {
...
+
+        fight.runningAway();
+
+        assertThat(fight.getMonsterCounter()).as("monsterCounter test to decrease in value").isEqualTo(0);
+        assertThat(fight.nextTurn()).as("There should be no next turn, because no more monster.").isFalse();
+
+
+
+
+
```
Comment (gergely-szanto-epam) [inline src/test/java/hu/dungeonhunter/TestFight.java]: remove these empty lines :)

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 3]
```diff
@@ -36,5 +36,34 @@ public void championDefeatTest() {
...
+    @Test
+    public void runningAwaySuccessfulEscapeTest(){
+        Fight fight = new Fight(1);
+        Champion highHpChampion = new Champion(10);
+        fight.setChampion(highHpChampion);
+
+        fight.runningAway();
+
+        assertThat(fight.getMonsterCounter()).as("monsterCounter test to decrease in value").isEqualTo(0);
+        assertThat(fight.nextTurn()).as("There should be no next turn, because no more monster.").isFalse();
```
Comment (gergely-szanto-epam) [inline src/test/java/hu/dungeonhunter/TestFight.java:62]: Also add assertion to check that champion is not defeated.

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 1 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: I corrected

File: src/test/java/hu/dungeonhunter/TestFight.java (hunk 2 above)
Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java]: ups :D 
corrected.

File: src/test/java/hu/dungeonhunter/TestFight.java (hunk 3 above)
Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java:62]: Thank you, it's done!

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 4]
```diff
@@ -37,4 +37,29 @@ public void championDefeatTest() {
...
+        assertThat(highHpChampion.isDefeat()).as("Champion should have been  no defeated.").isFalse();
```
Comment (gergely-szanto-epam) [inline src/test/java/hu/dungeonhunter/TestFight.java]:
```suggestion
        assertThat(highHpChampion.isDefeat()).as("Champion should have been not defeated.").isFalse();
```

Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java]: I corrected.

**Review by gergely-szanto-epam — APPROVED** [review]

**Pull request merged**